Doc/library/typing.rst

```diff
@@ -572,6 +572,37 @@ These can be used as types in annotations and do not support ``[]``.
    * Every type is compatible with :data:`Any`.
    * :data:`Any` is compatible with every type.
 
+.. data:: Never
+
+   The `bottom type <https://en.wikipedia.org/wiki/Bottom_type>`_,
+   a type that has no members.
+
+   This can be used to define a function that should never be
+   called, or a function that never returns::
+
+     from typing import Never
+
+     def never_call_me(arg: Never) -> None:
+         pass
+
+     def int_or_str(arg: int | str) -> None:
+         never_call_me(arg)  # type checker error
+         match arg:
+             case int():
+                 print("It's an int")
+             case str():
+                 print("It's a str")
+             case _:
+                 never_call_me(arg)  # ok, arg is of type Never
+
+     def stop() -> Never:
```

Comment:
Are we deprecating `NoReturn`? If not, in which cases are we saying `NoReturn` should still be used — surely this is a textbook example of a situation where the `NoReturn` name is more descriptive of the semantics?

Reply:
Good point, I'll remove this second example.

Reply:
Looks like the same example's on line 478 of `typing.py` as well (in the docstring of `Never`)

Reply:
Thanks, fixed that one too